src/test/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepTest.java

```diff
@@ -363,4 +380,95 @@ private void runAndContinue(JenkinsRule.WebClient webClient, WorkflowJob foo, St
         j.assertBuildStatus(Result.ABORTED, p.scheduleBuild2(0).get());
     }
 
+    @Issue("JENKINS-47699")
+    @Test
+    public void userScopedCredentials() throws Exception {
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        final User alpha = User.getOrCreateByIdOrFullName("alpha");
+        final String alphaSecret = "correct horse battery staple";
+        final String alphaId = registerUserSecret(alpha, alphaSecret);
+        final User beta = User.getOrCreateByIdOrFullName("beta");
+        final String betaSecret = "hello world bad password";
+        final String betaId = registerUserSecret(beta, betaSecret);
+        final User gamma = User.getOrCreateByIdOrFullName("gamma");
+        final String gammaSecret = "proton mass decay string";
+        final String gammaId = registerUserSecret(gamma, gammaSecret);
+        final User delta = User.getOrCreateByIdOrFullName("delta");
+        final String deltaSecret = "fundamental arithmetic theorem prover";
+        final String deltaId = registerUserSecret(delta, deltaSecret);
+
+        final WorkflowJob p = j.createProject(WorkflowJob.class);
+        p.addProperty(new ParametersDefinitionProperty(
+                new CredentialsParameterDefinition("deltaId", null, null, StringCredentialsImpl.class.getName(), true)
+        ));
+        p.setDefinition(new CpsFlowDefinition("node {\n" +
+                stringCredentialsInput("AlphaCreds", "alphaId") +
+                stringCredentialsInput("BetaCreds", "betaId") +
+                stringCredentialsInput("GammaCreds", "gammaId") +
+                "  withCredentials([\n" +
+                "      string(credentialsId: 'alphaId', variable: 'alphaSecret'),\n" +
```

Comment:
If the build was running with something like authorize-project enabled, would `withCredentials` still work if the authentication for the build was a different user than the user who owns the user the user-scope credentials that were submitted to the `input`? Just trying to understand how the lookup works for user-scope credentials.

Reply:
Based on some cursory review of `findCredentialsById()`, it looks like this would still work. The triggering user being different from the input steps is explicitly supported without authorize-project installed, and with that installed, it only adds more credentials to the candidate list, not removing them.

The only potential problematic scenario would be if someone is using authorize-project, has a hand-coded user-scoped credentials id in the pipeline, and they also have a credentials parameter or credentials input with the same name as that credentials id. In this scenario, the parameter name would take precedence over the user-scoped credential that was hard-coded.

Reply:
Yeah it's not like it would have worked before, I was just curious whether the lookup for user-scope credentials takes into account the current authentication or just checks everything available in CredentialsParameterBinder without regard to the current authentication.

Reply:
The binder keeps everything as CredentialsParameterBindings which associate which user provided the credential which allows us to look up user-scoped credentials for that authentication. Theoretically, this mechanism could be extended by a security realm to make this fancier.